# Innobyte services Intership task
## Amazon Sales report analysis
### Name : Vaibhav Agrawal

### Introduction
- In this report, we are analysing the amazon sales report data which containes the sales data of different amazon product to from 1 april to 1 july to provide useful business insights for expanding the business.
#### Libraries used
- numpy
- pandas
- seaborn
- matplotlib
#### About the features
##### Index 
- Unique identifier for each record.
- Integer
##### Order ID
- Unique identifier for each order.
- String
##### Date
- Date when the order was placed.
- Date format - yyyy-DD-MM.
##### Status
- Current status of the order.
- String
##### Fulfilment
- Method by which the order is fulfilled.
- String

##### Sales Channel
- Platform through which the order was made.
- String
##### Ship-Service-Level
- Level of shipping service chosen for the order.
- String
##### Category
- Product category of the ordered item.
- String
##### Size
- Size of the ordered item, if applicable.
- String
##### Courier Status
- Current status of the order with the courier.
- String
##### Currency
- Currency in which the order was placed.
- String
##### Amount
- Total amount of the order.
- Float
##### Ship-City
- City to which the order is shipped.
- String
##### Ship-State
- State to which the order is shipped.
- String
##### Ship-Postal-Code
- Postal code to which the order is shipped.
- String
##### Ship-Country
- Country to which the order is shipped.
- String
##### B2B
- Indicates if the order is a Business-to-Business order.
- Boolean
##### Fulfilled-By
- Entity responsible for fulfilling the order.
- String


# Importing Libraries

In [39]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

### Importing Dataset 

In [40]:
df = pd.read_csv('q.csv', encoding = 'latin1')

### Display the first few rows of the dataset

In [41]:
df.head()

index             Order ID      Date                        Status  \
0      0  405-8078784-5731545  04-30-22                     Cancelled   
1      1  171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   
2      2  404-0687676-7273146  04-30-22                       Shipped   
3      3  403-9615377-8133951  04-30-22                     Cancelled   
4      4  407-1069790-7240320  04-30-22                       Shipped   

  Fulfilment Sales Channel ship-service-level  Category Size Courier Status  \
0   Merchant     Amazon.in           Standard   T-shirt    S     On the Way   
1   Merchant     Amazon.in           Standard     Shirt  3XL        Shipped   
2     Amazon     Amazon.in          Expedited     Shirt   XL        Shipped   
3   Merchant     Amazon.in           Standard   Blazzer    L     On the Way   
4     Amazon     Amazon.in          Expedited  Trousers  3XL        Shipped   

   ...  currency  Amount    ship-city   ship-state ship-postal-code  \
0  ...       INR  647.62       MUMBAI  MAHARASHTRA         400081.0   
1  ...       INR  406.00    BENGALURU    KARNATAKA         560085.0   
2  ...       INR  329.00  NAVI MUMBAI  MAHARASHTRA         410210.0   
3  ...       INR  753.33   PUDUCHERRY   PUDUCHERRY         605008.0   
4  ...       INR  574.00      CHENNAI   TAMIL NADU         600073.0   

   ship-country    B2B  fulfilled-by New  PendingS  
0            IN  False     Easy Ship NaN       NaN  
1            IN  False     Easy Ship NaN       NaN  
2            IN   True           NaN NaN       NaN  
3            IN  False     Easy Ship NaN       NaN  
4            IN  False           NaN NaN       NaN  

[5 rows x 21 columns]

# Exploratory Data Analysis

### Display basic information about the dataset

In [42]:
print("\nBasic information about the dataset:")
print(df.info())


Basic information about the dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128976 entries, 0 to 128975
Data columns (total 21 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   index               128976 non-null  int64  
 1   Order ID            128976 non-null  object 
 2   Date                128976 non-null  object 
 3   Status              128976 non-null  object 
 4   Fulfilment          128976 non-null  object 
 5   Sales Channel       128976 non-null  object 
 6   ship-service-level  128976 non-null  object 
 7   Category            128976 non-null  object 
 8   Size                128976 non-null  object 
 9   Courier Status      128976 non-null  object 
 10  Qty                 128976 non-null  int64  
 11  currency            121176 non-null  object 
 12  Amount              121176 non-null  float64
 13  ship-city           128941 non-null  object 
 14  ship-state          128941 non-null  object 
 

### Display summary statistics of the dataset

In [43]:
print("\nSummary statistics of the dataset:")
print(df.describe())


Summary statistics of the dataset:
               index            Qty         Amount  ship-postal-code  New  \
count  128976.000000  128976.000000  121176.000000     128941.000000  0.0   
mean    64486.130427       0.904401     648.562176     463945.677744  NaN   
std     37232.897832       0.313368     281.185041     191458.488954  NaN   
min         0.000000       0.000000       0.000000     110001.000000  NaN   
25%     32242.750000       1.000000     449.000000     382421.000000  NaN   
50%     64486.500000       1.000000     605.000000     500033.000000  NaN   
75%     96730.250000       1.000000     788.000000     600024.000000  NaN   
max    128974.000000      15.000000    5584.000000     989898.000000  NaN   

       PendingS  
count       0.0  
mean        NaN  
std         NaN  
min         NaN  
25%         NaN  
50%         NaN  
75%         NaN  
max         NaN  


In [1]:
if 'date' in df.columns:
    df['Date'] = pd.to_datetime(df['Date'])

NameError: name 'df' is not defined

In [45]:
df.isna().sum()

index                      0
Order ID                   0
Date                       0
Status                     0
Fulfilment                 0
Sales Channel              0
ship-service-level         0
Category                   0
Size                       0
Courier Status             0
Qty                        0
currency                7800
Amount                  7800
ship-city                 35
ship-state                35
ship-postal-code          35
ship-country              35
B2B                        0
fulfilled-by           89713
New                   128976
PendingS              128976
dtype: int64

In [46]:
df = df.drop(columns = ['New', 'PendingS', 'index'])
df.head()

Order ID      Date                        Status Fulfilment  \
0  405-8078784-5731545  04-30-22                     Cancelled   Merchant   
1  171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   Merchant   
2  404-0687676-7273146  04-30-22                       Shipped     Amazon   
3  403-9615377-8133951  04-30-22                     Cancelled   Merchant   
4  407-1069790-7240320  04-30-22                       Shipped     Amazon   

  Sales Channel ship-service-level  Category Size Courier Status  Qty  \
0     Amazon.in           Standard   T-shirt    S     On the Way    0   
1     Amazon.in           Standard     Shirt  3XL        Shipped    1   
2     Amazon.in          Expedited     Shirt   XL        Shipped    1   
3     Amazon.in           Standard   Blazzer    L     On the Way    0   
4     Amazon.in          Expedited  Trousers  3XL        Shipped    1   

  currency  Amount    ship-city   ship-state  ship-postal-code ship-country  \
0      INR  647.62       MUMBAI  MAHARASHTRA          400081.0           IN   
1      INR  406.00    BENGALURU    KARNATAKA          560085.0           IN   
2      INR  329.00  NAVI MUMBAI  MAHARASHTRA          410210.0           IN   
3      INR  753.33   PUDUCHERRY   PUDUCHERRY          605008.0           IN   
4      INR  574.00      CHENNAI   TAMIL NADU          600073.0           IN   

     B2B fulfilled-by  
0  False    Easy Ship  
1  False    Easy Ship  
2   True          NaN  
3  False    Easy Ship  
4  False          NaN

In [47]:
df[df.duplicated()].shape

(959, 18)

In [48]:
df.drop_duplicates(inplace = True)

In [49]:
df[df['Qty'] == 0]

Order ID      Date     Status Fulfilment Sales Channel  \
0       405-8078784-5731545  04-30-22  Cancelled   Merchant     Amazon.in   
3       403-9615377-8133951  04-30-22  Cancelled   Merchant     Amazon.in   
8       407-5443024-5233168  04-30-22  Cancelled     Amazon     Amazon.in   
23      404-6019946-2909948  04-30-22  Cancelled   Merchant     Amazon.in   
29      404-5933402-8801952  04-30-22  Cancelled   Merchant     Amazon.in   
...                     ...       ...        ...        ...           ...   
128903  406-4220789-1106743  05-31-22  Cancelled     Amazon     Amazon.in   
128904  406-3923120-4345139  05-31-22  Cancelled     Amazon     Amazon.in   
128908  403-6757403-6097100  05-31-22  Cancelled     Amazon     Amazon.in   
128959  408-9513596-4393945  05-31-22  Cancelled     Amazon     Amazon.in   
128968  404-5182288-1653947  05-31-22  Cancelled     Amazon     Amazon.in   

       ship-service-level Category Size Courier Status  Qty currency  Amount  \
0                Standard  T-shirt    S     On the Way    0      INR  647.62   
3                Standard  Blazzer    L     On the Way    0      INR  753.33   
8               Expedited  T-shirt  3XL      Cancelled    0      NaN     NaN   
23               Standard  T-shirt    M     On the Way    0      INR  570.48   
29               Standard    Shirt  3XL     On the Way    0      NaN     NaN   
...                   ...      ...  ...            ...  ...      ...     ...   
128903          Expedited  Blazzer    L      Cancelled    0      NaN     NaN   
128904          Expedited    Shirt    M      Cancelled    0      NaN     NaN   
128908          Expedited  T-shirt  3XL      Cancelled    0      NaN     NaN   
128959          Expedited  T-shirt    L      Cancelled    0      NaN     NaN   
128968          Expedited    Shirt   XS      Cancelled    0      NaN     NaN   

            ship-city      ship-state  ship-postal-code ship-country    B2B  \
0              MUMBAI     MAHARASHTRA          400081.0           IN  False   
3          PUDUCHERRY      PUDUCHERRY          605008.0           IN  False   
8           HYDERABAD       TELANGANA          500008.0           IN  False   
23               pune     MAHARASHTRA          411044.0           IN  False   
29           GUWAHATI           ASSAM          781003.0           IN  False   
...               ...             ...               ...          ...    ...   
128903      ANANTAPUR  ANDHRA PRADESH          515001.0           IN  False   
128904      ANANTAPUR  ANDHRA PRADESH          515001.0           IN  False   
128908  GREATER NOIDA   UTTAR PRADESH          201306.0           IN  False   
128959      Bengaluru       KARNATAKA          560037.0           IN  False   
128968        Kolkata     WEST BENGAL          700040.0           IN  False   

       fulfilled-by  
0         Easy Ship  
3         Easy Ship  
8               NaN  
23        Easy Ship  
29        Easy Ship  
...             ...  
128903          NaN  
128904          NaN  
128908          NaN  
128959          NaN  
128968          NaN  

[12543 rows x 18 columns]

In [50]:
df.drop(df[df['Qty'] == 0].index, inplace=True)
df

Order ID      Date                        Status  \
1       171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   
2       404-0687676-7273146  04-30-22                       Shipped   
4       407-1069790-7240320  04-30-22                       Shipped   
5       404-1490984-4578765  04-30-22                       Shipped   
6       408-5748499-6859555  04-30-22                       Shipped   
...                     ...       ...                           ...   
128971  406-6001380-7673107  05-31-22                       Shipped   
128972  402-9551604-7544318  05-31-22                       Shipped   
128973  407-9547469-3152358  05-31-22                       Shipped   
128974  402-6184140-0545956  05-31-22                       Shipped   
128975  408-7436540-8728312  05-31-22                       Shipped   

       Fulfilment Sales Channel ship-service-level  Category Size  \
1        Merchant     Amazon.in           Standard     Shirt  3XL   
2          Amazon     Amazon.in          Expedited     Shirt   XL   
4          Amazon     Amazon.in          Expedited  Trousers  3XL   
5          Amazon     Amazon.in          Expedited   T-shirt   XL   
6          Amazon     Amazon.in          Expedited   T-shirt    L   
...           ...           ...                ...       ...  ...   
128971     Amazon     Amazon.in          Expedited     Shirt   XL   
128972     Amazon     Amazon.in          Expedited   T-shirt    M   
128973     Amazon     Amazon.in          Expedited   Blazzer  XXL   
128974     Amazon     Amazon.in          Expedited   T-shirt   XS   
128975     Amazon     Amazon.in          Expedited   T-shirt    S   

       Courier Status  Qty currency  Amount    ship-city     ship-state  \
1             Shipped    1      INR   406.0    BENGALURU      KARNATAKA   
2             Shipped    1      INR   329.0  NAVI MUMBAI    MAHARASHTRA   
4             Shipped    1      INR   574.0      CHENNAI     TAMIL NADU   
5             Shipped    1      INR   824.0    GHAZIABAD  UTTAR PRADESH   
6             Shipped    1      INR   653.0   CHANDIGARH     CHANDIGARH   
...               ...  ...      ...     ...          ...            ...   
128971        Shipped    1      INR   517.0    HYDERABAD      TELANGANA   
128972        Shipped    1      INR   999.0     GURUGRAM        HARYANA   
128973        Shipped    1      INR   690.0    HYDERABAD      TELANGANA   
128974        Shipped    1      INR  1199.0        Halol        Gujarat   
128975        Shipped    1      INR   696.0       Raipur   CHHATTISGARH   

        ship-postal-code ship-country    B2B fulfilled-by  
1               560085.0           IN  False    Easy Ship  
2               410210.0           IN   True          NaN  
4               600073.0           IN  False          NaN  
5               201102.0           IN  False          NaN  
6               160036.0           IN  False          NaN  
...                  ...          ...    ...          ...  
128971          500013.0           IN  False          NaN  
128972          122004.0           IN  False          NaN  
128973          500049.0           IN  False          NaN  
128974          389350.0           IN  False          NaN  
128975          492014.0           IN  False          NaN  

[115474 rows x 18 columns]

In [51]:
df.isna().sum()

Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Category                  0
Size                      0
Courier Status            0
Qty                       0
currency                124
Amount                  124
ship-city                27
ship-state               27
ship-postal-code         27
ship-country             27
B2B                       0
fulfilled-by          83341
dtype: int64

In [52]:
df.dropna(subset = ['Amount', 'currency','ship-city', 'ship-state', 'ship-postal-code', 'ship-country'], inplace = True)
df.head()

Order ID      Date                        Status Fulfilment  \
1  171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   Merchant   
2  404-0687676-7273146  04-30-22                       Shipped     Amazon   
4  407-1069790-7240320  04-30-22                       Shipped     Amazon   
5  404-1490984-4578765  04-30-22                       Shipped     Amazon   
6  408-5748499-6859555  04-30-22                       Shipped     Amazon   

  Sales Channel ship-service-level  Category Size Courier Status  Qty  \
1     Amazon.in           Standard     Shirt  3XL        Shipped    1   
2     Amazon.in          Expedited     Shirt   XL        Shipped    1   
4     Amazon.in          Expedited  Trousers  3XL        Shipped    1   
5     Amazon.in          Expedited   T-shirt   XL        Shipped    1   
6     Amazon.in          Expedited   T-shirt    L        Shipped    1   

  currency  Amount    ship-city     ship-state  ship-postal-code ship-country  \
1      INR   406.0    BENGALURU      KARNATAKA          560085.0           IN   
2      INR   329.0  NAVI MUMBAI    MAHARASHTRA          410210.0           IN   
4      INR   574.0      CHENNAI     TAMIL NADU          600073.0           IN   
5      INR   824.0    GHAZIABAD  UTTAR PRADESH          201102.0           IN   
6      INR   653.0   CHANDIGARH     CHANDIGARH          160036.0           IN   

     B2B fulfilled-by  
1  False    Easy Ship  
2   True          NaN  
4  False          NaN  
5  False          NaN  
6  False          NaN

In [53]:
df.isna().sum()

Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Category                  0
Size                      0
Courier Status            0
Qty                       0
currency                  0
Amount                    0
ship-city                 0
ship-state                0
ship-postal-code          0
ship-country              0
B2B                       0
fulfilled-by          83200
dtype: int64

In [54]:
df.to_csv('updated_amazon_sales.csv',index=False)

# Business Insights

#### Total Sales Amount: The total sales amount is ₹78,590,170.25.
#### Average Sales Amount: The average sales amount is ₹648.56.
#### Sales Trends: The sales trends show a consistent pattern over time with some peaks, indicating periods of higher sales activity.
#### Most Sold Products: The most sold products are in the 'Category' column, with the top categories being displayed in the bar chart. This indicates which product categories are the most popular among customers.
#### Fulfillment Analysis: The fulfillment analysis shows the distribution of orders by fulfillment type, with the majority being fulfilled by 'Easy Ship'. This indicates the preferred fulfillment method.
#### Customer Segmentation: Customer segmentation reveals that the average order value per customer is ₹653.67. The distribution of the number of orders and total spending per customer provides insights into customer purchasing behavior.
#### Geographical Analysis: The geographical analysis highlights the top cities and states by the number of orders, with the top cities and states being displayed in the bar charts. This indicates the regions with the highest sales activity.


# Recommendations 

#### Optimize Inventory Management:
- Focus on stocking the most popular product categories identified in the product analysis to ensure availability and meet customer demand.
- Monitor sales trends to anticipate periods of higher sales activity and adjust inventory levels accordingly.

#### Enhance Fulfillment Efficiency:
- Since 'Easy Ship' is the preferred fulfillment method, consider expanding its use to improve delivery times and customer satisfaction.
- Evaluate the performance of other fulfillment methods and identify areas for improvement to ensure timely and efficient order delivery.

#### Targeted Marketing Campaigns:
- Use the insights from customer segmentation to create targeted marketing campaigns aimed at high-value customers and frequent buyers.
- Develop personalized promotions and offers based on customer purchasing behavior to increase customer loyalty and repeat purchases.

#### Geographical Expansion:
- Focus marketing efforts and promotional activities in the top-performing cities and states identified in the geographical analysis to further boost sales in these regions.
- Explore opportunities to expand into regions with lower sales activity by understanding the local market needs and preferences.

#### Improve Customer Experience:
- Analyze customer feedback and reviews to identify common pain points and areas for improvement in the shopping experience.
- Implement measures to address customer concerns, such as improving product quality, enhancing website usability, and providing excellent customer support.

#### Leverage Data Analytics:
- Continuously monitor and analyze sales data to identify emerging trends and make data-driven decisions.
- Use advanced analytics techniques, such as predictive modeling, to forecast future sales and optimize business strategies.

#### By implementing these recommendations, the business can optimize its sales strategies, improve customer satisfaction, and enhance overall performance, leading to increased revenue and growth.